platforms/common/cpp/core/Utf8.cpp

```diff
@@ -17,27 +17,89 @@
 #import <UIKit/UIKit.h>
 #endif
 
+// We store here compiler flag about which logic of utf8 decode we should use:
+//  false - usual way with loos real code for 3 bytes codes,
+//  true - with decoding 3 bytes codes into UTF-16 codes (pairs of 2 bytes codes).
+bool is_utf8_js_style = false;
+
 /* Character encoding */
 
-static inline int encode(uint8_t *out, uint16_t c) {
-    if (c <= 0x7F)
+static inline unsigned encode(uint16_t c, uint16_t next_c, bool is_last, uint8_t *out, int &cnt)
```

Comment:
It worth removing `inline`?

Reply:
Yes, `encode` function becomes a bit larger and `inline` looks not good now. Will remove it with next commit.